modules/dists/CyclicDist.chpl

```diff
@@ -512,6 +527,13 @@ proc CyclicDom.setup() {
   }
 }
 
+proc CyclicDom.dsiDestroyDom() {
+    coforall localeIdx in dist.targetLocDom {
+      on dist.targetLocs(localeIdx) do
+        delete locDoms(localeIdx);
+    }
+}
```

Comment:
I think it's reasonable/right that each distribution needs to specify how to delete classes it created.